src/prefect/infrastructure/docker.py

```diff
@@ -268,8 +268,12 @@ def _create_and_start_container(self) -> str:
 
     def _get_image_and_tag(self) -> Tuple[str, Optional[str]]:
         parts = self.image.split(":")
-        image = parts.pop(0)
-        tag = parts[0] if parts else None
+        try:
+            image = ':'.join(parts[:2]) if '/' in parts[1] else parts[0]
+            tag = parts[2] if '/' in parts[1] else parts[1]
+        except IndexError:
+            image= self.image
+            tag = None
```

Comment:
I'm having a hard time following this code now and I'm worried it doesn't account for all of the cases we'll encounter. I think we should introduce a utility in `prefect.docker` called `parse_image_tag` that is reused here. We should include a docstring that explains the expected format and output. It looks like there's some valuable information in this thread https://stackoverflow.com/questions/42115777/parsing-docker-image-tag-into-component-parts

cc @chrisguidry

Reply:
I wrote a more legible version of the function. The functionallity is nearly the same. Your original script could cope with more than on tag and my old version could do that to (by only using the first tag). Since that is not really format compliant, I now through an error in these cases.